src/reactnativeapp/README.md

```diff
@@ -0,0 +1,86 @@
+# Example React Native app
+
+This was created using [`npx create-expo-app@latest`](https://reactnative.dev/docs/environment-setup#start-a-new-react-native-project-with-expo)
+
+Content was taken from the web app example in src/frontend and modified to work
+in a React Native environment.
+
+## Get started
+
+Start the OpenTelemetry demo:
+
+```bash
+cd ../..
+make start # or start-minimal
+```
+
+## Building the app
+
+Unlike the other components under src/ which run within docker containers this
+app must be built and then run on a mobile simulator on your machine or a physical
+device. If this is your first time running a React Native app then in order to
+execute the steps under "Build on your host machine" you will need to setup your
+local environment for Android or iOS development or both following
+[this guide](https://reactnative.dev/docs/set-up-your-environment). Alternatively
+for Android you can instead follow the steps under "Build within a container" to
+leverage a container to build the app's apk for you.
+
+### Build on your host machine
+
+Build and run the React Native app for a given platform by running these make targets
+from the project root:
+
+```bash
+make reactnative-run-android
+```
+
+Or
+
+```bash
+make reactnative-run-ios
```

Comment:
When running this I get the following error:

```
...
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:iOS Simulator, id:FF9D57D3-43CB-42B6-8100-A49B4BEAB05D, OS:18.1, name:iPhone 16 Pro }
{ platform:iOS Simulator, id:FF9D57D3-43CB-42B6-8100-A49B4BEAB05D, OS:18.1, name:iPhone 16 Pro }
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution [CP-User]\ Generate\ Specs /Users/juliano.costa/Library/Developer/Xcode/DerivedData/reactnativeapp-hfjcezijrrdpihfcaelpfppoavlp/Build/Intermediates.noindex/Pods.build/Release-iphonesimulator/React-Codegen.build/Script-46EB2E00019480.sh (in target 'React-Codegen' from project 'Pods')
	Building workspace reactnativeapp with scheme reactnativeapp and configuration Release
(2 failures)
Build logs written to /Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/.expo/xcodebuild.log
make: *** [reactnative-run-ios] Error 1
```

The log file has 26700 lines. The `error` word is found 10113 times 🙃 .

I also get a lot of warnings like this:

```
/Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:146:13: warning: parameter 'progress' not found in the function declaration [-Wdocumentation]
  146 |    * @param progress
      |             ^~~~~~~~
/Users/juliano.costa/workspace/opentelemetry-demo/src/reactnativeapp/ios/Pods/Headers/Public/React-graphics/react/renderer/graphics/Transform.h:146:13: note: did you mean 'animationProgress'?
  146 |    * @param progress
      |             ^~~~~~~~
      |             animationProgress
```

The `npm run ios` command also fails.

Reply:
@jpmunz I forgot to mention that I'm still able to run the project as I did before.

```
cd src/reactnativeapp
npm install
npm start
```

Then I go to XCode, build and run.

The main point of my comment was to say that we need to add those steps into the README, or change the current `make` command to run those steps.

Reply:
for `npm install` that I think should already be part of the make, the target has a dependency on the `node_modules` folder being there:
```
.PHONY: reactnative-run-ios
reactnative-run-ios: src/reactnativeapp/node_modules
  npm --prefix src/reactnativeapp run ios
```

which should then trigger the npm install:
```
src/reactnativeapp/node_modules:
  npm --prefix src/reactnativeapp install
```

`npm start` should only be required when doing a Debug build to serve up the JS bundle, to simplify things I had changed the commands to do release builds where the JS bundle is included as part of the app build. When you were able to build from XCode was that doing a Debug or Release configuration?

If it continues to cause problems I can switch things back to doing debug builds by default and add those steps + going to XCode as a troubleshooting fallback to the README (`npm run ios-debug` alone in theory should be enough as it both spins up the dev server and does the build in the same command)

Reply:
I'm just trying to follow the README.
When it doesn't work, I go to XCode to see if things still works.

In Xcode I can only run the app when I have the debug server running, so I guess I'm running debug mode.
But TBH I haven't configured anything, I just go there and hit play.

Reply:
ya would be the debug configuration by default in XCode, would have to run through the steps in those screenshots I posted above to swap it to the Release configuration

Reply:
@julianocosta89 maybe best if we could do a screenshare to go over the issues you're hitting so we can narrow down the improvements we should make to the README / commands? Could maybe use the Demo SIG meeting slot this Wednesday if the agenda is open, I'm also happy to schedule something separately if you're available

Reply:
Sounds good, I'll ping you on Slack